cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1314,8 +1314,94 @@ if(vc AND NOT Vc_FOUND)
 endif()
 
 if(Vc_FOUND)
-	# Missing from VcConfig.cmake
-	set(Vc_INCLUDE_DIRS ${Vc_INCLUDE_DIR})
+  # Missing from VcConfig.cmake
+  set(Vc_INCLUDE_DIRS ${Vc_INCLUDE_DIR})
+endif()
+
+#---Check for VecCore--------------------------------------------------------------------
+if(veccore AND builtin_vc)
+  message(WARNING "Vc is not relocatable, so 'builtin_vc' requires 'builtin_veccore' to set up Vc properly.")
+  set(builtin_veccore ON CACHE BOOL "" FORCE)
+endif()
+
+if(builtin_veccore)
+  unset(VecCore_FOUND)
+  unset(VecCore_FOUND CACHE)
+  set(veccore ON CACHE BOOL "" FORCE)
+elseif(veccore)
+  if(fail-on-missing)
+    find_package(VecCore 0.3.1 CONFIG QUIET REQUIRED)
+  else()
+    find_package(VecCore 0.3.1 CONFIG QUIET)
+    if(NOT VecCore_FOUND)
+      message(STATUS "VecCore not found, support for it disabled.")
+      message(STATUS "Please enable the option 'builtin_veccore' to build VecCore internally.")
+      set(veccore OFF CACHE BOOL "" FORCE)
+    endif()
+  endif()
+endif()
+
+if(veccore AND NOT VecCore_FOUND)
+  set(VecCore_VERSION "0.3.1")
+  set(VecCore_PROJECT "VecCore-${VecCore_VERSION}")
+  set(VecCore_SRC_URI "${lcgpackages}/veccore_v${VecCore_VERSION}.tar.gz")
+  set(VecCore_SRC_MD5 "1afbb78f423bfa15a14d00843d1941b3")
+  set(VecCore_DESTDIR "${CMAKE_BINARY_DIR}/externals/install/${VecCore_PROJECT}")
+  set(VecCore_ROOTDIR "${VecCore_DESTDIR}/${CMAKE_INSTALL_PREFIX}")
+
+  if(builtin_vc)
+    set(Vc_LIBNAME "${CMAKE_STATIC_LIBRARY_PREFIX}Vc${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(Vc_LIBRARY "${VecCore_ROOTDIR}/${_LIBDIR_DEFAULT}/${Vc_LIBNAME}")
+  endif()
+
+  ExternalProject_Add(${VecCore_PROJECT}
+    PREFIX externals
+    URL "${VecCore_SRC_URI}"
+    URL_MD5 ${VecCore_SRC_MD5}
+    BUILD_IN_SOURCE 0
+    BUILD_BYPRODUCTS ${Vc_LIBRARY}
+    LOG_DOWNLOAD 1 LOG_CONFIGURE 1 LOG_BUILD 1 LOG_INSTALL 1
+    CMAKE_ARGS -G ${CMAKE_GENERATOR}
+               -DBUILD_TESTING=OFF -DVC=${vc} -DBUILD_VC=${builtin_vc}
```

Comment:
I thought that since it's external, and already tested with GitLab CI and Travis CI, it would be better to turn off testing here to avoid problems if enabled by accident in unsupported archtectures in ROOT tests. However, if we do want to make sure that it works in all those environments, we can certainly turn this on here conditionally (i.e. when ROOT itself is being tested).